---
title: "simulation_outline"
author: "Jennifer Blanc"
date: "9/18/2020"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
library(data.table)
library(ggplot2)
```

## Causal Diagram 

```{r, out.height="100%", out.width="100%", echo=FALSE}
knitr::include_graphics("assets/causal_diagram.jpeg", error = FALSE)
```

## Simulations 1 

**Simulate Data** 
Randomly choose derived status of effect allele
```{r}
L <- 100
D <- rbinom(L, 1, 0.5)
```

Set $\gamma_{s|D} = 0.5$ and generate frequency difference w/ error    
```{r}
gamma_s_D <- 0.5
s <- gamma_s_D * D + rnorm(L, 0, 0.1)
plot(s,D)
```

Set $\gamma_{B|S} = 3$ and generate estimated effect sizes w/ error
```{r}
gamma_B_s <- 3
Bhat <- gamma_B_s * s + rnorm(L, 0, 0.1)
plot(s, Bhat)
```

**Fit model**  

Regress $\vec{s} \sim \vec{D}$ to estimate $\hat{\gamma_{s|D}}$
```{r}
stg1 <- lm(s ~ D)
stg1
hat_gamma_s_D <- stg1$coefficients[2]
```

Regress $\hat{\vec{\beta}} \sim \hat{\gamma}_{s|D}\vec{D}$ to estimate $\hat{\gamma}_{B|s}$
```{r}
s_hat <- as.numeric(hat_gamma_s_D)*D
stg2 <- lm(Bhat ~ s_hat)
stg2
```

Our estimate of $\gamma_{B|S}$ is: 
```{r}
stg2$coefficients[2]
```

## Exploring SNP loadings 

**Single Population**  

Simulate single population and output genotype matrix 
```{r, eval=FALSE}
ts = msprime.simulate(sample_size=1000,Ne=5000,length=1e6, recombination_rate=1e-8,
                         mutation_rate=1e-8, random_seed = 12)
G = ts.genotype_matrix()
```

Read in and standardize genotype matrix 
```{r}
G <- fread("../data/simple_simulations/G_onepop.txt")
G <- scale(t(G))
dim(G)
```

Singular value decompostion of the genotype matrix    
```{r}
decomp <- svd(G)
u <- decomp$u
v <- decomp$v
```

Make PCA plot 
```{r}
qplot(u[,1], u[,2])
```

Plot loading means 
```{r}
load_mean <- colMeans(v)
qplot(load_mean[1:20], seq(1,20)) + geom_vline(xintercept = 0, col = "red") + xlab("Average Loading") + ylab("PC")
```

**Population Split - same size** 

Simulate 2 population with same sizes 
```{r, eval=FALSE}
N_A = 2000
N_B = 2000
```

Read in and standardize genotype matrix 
```{r}
G <- fread("../data/simple_simulations/G_split_same.txt")
G <- scale(t(G))
dim(G)
```

Singular value decompostion of the genotype matrix    
```{r}
decomp <- svd(G)
u <- decomp$u
v <- decomp$v
```

Make PCA plot 
```{r}
labels = c(rep("A", 500), rep("B", 500))
qplot(u[,1], u[,2], col = labels)
```

Plot loading means 
```{r}
load_mean <- colMeans(v)
qplot(load_mean[1:20], seq(1,20)) + geom_vline(xintercept = 0, col = "red") + xlab("Average Loading") + ylab("PC")
```

**Population Split - A > B** 

Simulate 2 populations 
```{r, eval=FALSE}
N_A = 8000
N_B = 2000
```

Read in and standardize genotype matrix 
```{r}
G <- fread("../data/simple_simulations/G_split_Abig.txt")
G <- scale(t(G))
dim(G)
```

Singular value decompostion of the genotype matrix    
```{r}
decomp <- svd(G)
u <- decomp$u
v <- decomp$v
```

Make PCA plot 
```{r}
labels = c(rep("A", 500), rep("B", 500))
qplot(u[,1], u[,2], col = labels)
```

Plot loading means 
```{r}
load_mean <- colMeans(v)
qplot(load_mean[1:20], seq(1,20)) + geom_vline(xintercept = 0, col = "red") + xlab("Average Loading") + ylab("PC")
```


**Population Split - B > A** 

Simulate 2 populations 
```{r, eval=FALSE}
N_A = 2000
N_B = 8000
```

Read in and standardize genotype matrix 
```{r}
G <- fread("../data/simple_simulations/G_split_Bbig.txt")
G <- scale(t(G))
dim(G)
```

Singular value decompostion of the genotype matrix    
```{r}
decomp <- svd(G)
u <- decomp$u
v <- decomp$v
```

Make PCA plot 
```{r}
labels = c(rep("A", 500), rep("B", 500))
qplot(u[,1], u[,2], col = labels)
```


Plot loading means 
```{r}
load_mean <- colMeans(v)
qplot(load_mean[1:20], seq(1,20)) + geom_vline(xintercept = 0, col = "red") + xlab("Average Loading") + ylab("PC")
```


**OOA YRI and CEU**  
Read in and standardize genotype matrix 
```{r}
G <- fread("../data/simple_simulations/G_small_1.txt")
G <- scale(t(G))
dim(G)
```

Singular value decompostion of the genotype matrix    
```{r}
decomp <- svd(G)
u <- decomp$u
v <- decomp$v
```

Make PCA plot 
```{r}
labels = c(rep("YRI", 1000), rep("CEU", 1000))
qplot(u[,1], u[,2], col = labels)
```

Plot loading means 
```{r}
load_mean <- colMeans(v)
qplot(load_mean[1:20], seq(1,20)) + geom_vline(xintercept = 0, col = "red") + xlab("Average Loading") + ylab("PC")
```



