---
title: "simulation_outline"
author: "Jennifer Blanc"
date: "9/18/2020"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
library(data.table)
library(ggplot2)
```

## Causal Diagram 

```{r, out.height="100%", out.width="100%", echo=FALSE}
knitr::include_graphics("assets/causal_diagram.jpeg", error = FALSE)
```

## Simulations 1 

**Simulate Data** 
Randomly choose derived status of effect allele
```{r}
L <- 100
D <- rbinom(L, 1, 0.5)
```

Set $\gamma_{s|D} = 0.5$ and generate frequency difference w/ error    
```{r}
gamma_s_D <- 0.5
s <- gamma_s_D * D + rnorm(L, 0, 0.1)
plot(s,D)
```

Set $\gamma_{B|S} = 3$ and generate estimated effect sizes w/ error
```{r}
gamma_B_s <- 3
Bhat <- gamma_B_s * s + rnorm(L, 0, 0.1)
plot(s, Bhat)
```

**Fit model**  

Regress $\vec{s} \sim \vec{D}$ to estimate $\hat{\gamma_{s|D}}$
```{r}
stg1 <- lm(s ~ D)
stg1
hat_gamma_s_D <- stg1$coefficients[2]
```

Regress $\hat{\vec{\beta}} \sim \hat{\gamma}_{s|D}\vec{D}$ to estimate $\hat{\gamma}_{B|s}$
```{r}
s_hat <- as.numeric(hat_gamma_s_D)*D
stg2 <- lm(Bhat ~ s_hat)
stg2
```

Our estimate of $\gamma_{B|S}$ is: 
```{r}
stg2$coefficients[2]
```

## Exploring SNP loadings 

First I simulated 2000 individuals (1000 YRI, 1000 CEU) and output the Genotype matrix and derived allele frequency of varriants in each population.   

Read in genotype matrix 
```{r}
G <- fread("../data/simple_simulations/G_small_1.txt")
dim(G)
```

Singular value decompostion of the genotype matrix    
```{r}
decomp <- svd(G, nu = 20)
u <- decomp$u
v <- decomp$v
```

Make PCA plot  
```{r}
labels <- as.factor(c(rep("YRI",1000), rep("CEU",1000)))
qplot(v[,1], v[,2], col = labels)
```

Double check that we have the correct labels by checking the number of segregating sites in each population - YRI should have more segregating sites   
```{r}
G_YRI <- G[,1:1000]
G_CEU <- G[,1001:2000]

sums <- rowSums(G_YRI)
length(sums[sums != 0])

sums <- rowSums(G_CEU)
length(sums[sums != 0])
```

Calculate the average SNP loading (across all variants) for the top 20 PCs  
```{r}
dim(u)
load_mean <- colMeans(u)
qplot(load_mean[1:20], seq(1,20)) + geom_vline(xintercept = 0, col = "red") + xlab("Average Loading") + ylab("PC")
```



